### ISSUE TITLE
Strange behavior of FormGroup in custom control

### ISSUE THREAD

**Issue by VitalyV** [angular/angular#50970]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

I created my own custom control, which slightly changes the behavior of DatePicker from Angular material.
Inside the custom component, I use FormGroup to store the Start and End controls.
The main problem arose in the `writeValue` method.
Inside this method there is a code that sets a new state for the internal form and when calling `patchValue` with `emitEvent: false` parameter. But it doesn't work as expected.

### Reproduction

StackBlitz link: https://stackblitz.com/edit/stackblitz-starters-lsrf8q
Steps to reproduce:
1. Select any date in Datepicker
2. Click the `Set custom date range` button

### Expected Behavior

I expect that when clicking on the button in the console, only one line `#-write_value` will be output

### Actual Behavior

In addition to the string `#-write_value`, the string `#-selected_date` is also output, which is very strange, because I set the flag `emitEvent: false`

Why is this happening? What am I missing out on?

### Environment

- Angular: 16.1.3
- CDK/Material: 16.1.3
- Browser(s): Chrome 114.0.5735.199
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 11

**Comment by VitalyV** [comment]
Ok. Thanks

**Comment by VitalyV** [comment]
@wagnermaciel Was it possible to see an example?

**Comment by VitalyV** [comment]
The problem that is described above

**Comment by wagnermaciel** [comment]
Not exactly. When I tried the steps to reproduce, I wasn't able to get the `#-selected_date` in the console

<img width="534" alt="Screenshot 2023-07-11 at 1 23 27 PM" src="https://github.com/angular/angular/assets/25158423/de954b1c-3d72-4023-a223-7e31f42d8d5d">

**Comment by VitalyV** [comment]
![image](https://github.com/angular/angular/assets/6535549/627d2564-0410-4da1-9fd0-b47a0d682be7)

**Comment by VitalyV** [comment]
@wagnermaciel What should I do? They close this task there and reopen it here.

**Comment by wagnermaciel** [comment]
I already transferred the issue to angular/angular so there's nothing you'd need to do here. I can see that you closed this issue as completed yesterday. Was that intentional?

**Comment by VitalyV** [comment]
No, it was by accident, but I did not reopen it, because I realized that the issue is not related to angular material.

**Comment by wagnermaciel** [comment]
The issue has already been transferred to the angular/angular repo

**Comment by JeanMeche** [comment]
I'd tend to disagree on this. Forms work as expected here, it's the DatePicker that fires the value change event. 

`MatDatepickerInputBase.writeValue` invokes `onValidatorChange()` via `_assignValueProgrammatically()`.
